Storage/transaction_result.h

```diff
@@ -33,7 +33,12 @@ namespace storage {
 template <typename T>
 class TransactionResult : public OptionalResult<T> {
  public:
+#ifndef CURRENT_WINDOWS
   TransactionResult() = delete;
+#else
+  // MSVS needs this for `_Associated_state<>`, required by `std::promise<>`.
```

Comment:
Yes, but given we build & run on non-Windows every single time, I don't think this "sacrifice" is in any way big.